### PR TITLE
Fix "This should never happen"

### PR DESCRIPTION
**Pull request by vieiralucas**

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -51,11 +51,10 @@ stringP :: String -> Parser String
 stringP = sequenceA . map charP
 
 jsonBool :: Parser JsonValue
-jsonBool = f <$> (stringP "true" <|> stringP "false")
-   where f "true"  = JsonBool True
-         f "false" = JsonBool False
-         -- This should never happen
-         f _       = undefined
+jsonBool = jsonTrue <|> jsonFalse
+   where mkJsonBool str b = const (JsonBool b) <$> stringP str
```
Comment (rexim) [inline Main.hs]: I don't see any value in `mkJsonBool`, please inline it.

Comment (rexim) [inline Main.hs]: `const` + `<$>` is such a common pattern that there is a shorthand for it `<$`. Try `JsonBool b <$ stringP str`

**Review by rexim — CHANGES_REQUESTED** [review]

File: Main.hs (hunk 1 above)
Comment (vieiralucas) [reply Main.hs]: That's pretty cool

**Comment by vieiralucas** [comment]
@rexim made the requested changes

**Comment by rexim** [comment]
@vieiralucas looks good to me! :+1: Thank you for your contribution!

**Pull request merged**